allennlp/data/semparse/worlds/nlvr_world.py

```diff
@@ -70,15 +71,14 @@ class Box:
     ----------
     objects_list : ``List[JsonDict]``
         List of objects in the box, as given by the json file.
-    name : ``str`` (optional)
-        Optionally specify a string representation. It could be any unique string. If not
-        specified, we will use the list of object names.
     """
+    _id = 1
+
     def __init__(self,
-                 objects_list: List[JsonDict],
-                 name: str = None) -> None:
-        self.objects = set([Object(object_dict) for object_dict in objects_list])
-        self._name = name or str([str(obj) for obj in objects_list])
+                 objects_list: List[JsonDict]) -> None:
+        self._name = f"box {Box._id}"
+        Box._id += 1
```

Comment:
This is what's making your test fail.  Can you take the box id as an input to this function (where the values will always be 0, 1, or 2), instead of a global id?  Actually, it looks like that's what this was like before this change.  Why can't you just pass the box name as it was before this PR to the object?  Why also change the box name in this way?  Seems unnecessary, and it breaks things.

Reply:
Thanks! Fixed it.